src/Common/src/Interop/OSX/Interop.CoreFoundation.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using CFStringRef = System.IntPtr;
+using CFArrayRef = System.IntPtr;
+using FSEventStreamRef = System.IntPtr;
+using size_t = System.IntPtr;
+using FSEventStreamEventId = System.UInt64;
+using CFTimeInterval = System.Double;
+using CFRunLoopRef = System.IntPtr;
+
+internal static partial class Interop
+{
+    internal static partial class CoreFoundation
+    {
+        /// <summary>
+        /// This constant specifies that we want to use the default Run mode for the thread's Run loop.
+        /// </summary>
+        /// <remarks>
+        /// For more information, see the Apple documentation: https://developer.apple.com/library/mac/documentation/CoreFoundation/Reference/CFRunLoopRef/index.html
+        /// </remarks>
+        internal static IntPtr kCFRunLoopDefaultMode = CFStringCreateWithCString(IntPtr.Zero, "kCFRunLoopDefaultMode", 0);
+
+        /// <summary>
+        /// This is the default system path for the CoreFoundation library that PInvoke will use
+        /// </summary>
+        private const string CoreFoundationLibrary = "/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation";
+
+        /// <summary>
+        /// Tells the OS what encoding the passed in String is in. These come from the CFString.h header file in the CoreFoundation framework.
+        /// </summary>
+        internal enum CFStringBuiltInEncodings : uint
+        {
+            kCFStringEncodingMacRoman       = 0,
+            kCFStringEncodingWindowsLatin1  = 0x0500,
+            kCFStringEncodingISOLatin1      = 0x0201,
+            kCFStringEncodingNextStepLatin  = 0x0B01,
+            kCFStringEncodingASCII          = 0x0600,
+            kCFStringEncodingUnicode        = 0x0100,
+            kCFStringEncodingUTF8           = 0x08000100,
+            kCFStringEncodingNonLossyASCII  = 0x0BFF,
+
+            kCFStringEncodingUTF16 = 0x0100,
+            kCFStringEncodingUTF16BE = 0x10000100,
+            kCFStringEncodingUTF16LE = 0x14000100,
+            kCFStringEncodingUTF32 = 0x0c000100,
+            kCFStringEncodingUTF32BE = 0x18000100,
+            kCFStringEncodingUTF32LE = 0x1c000100
+        }
+
+        /// <summary>
+        /// Creates a CFStringRef from a 8-bit String object. Follows the "Create Rule" where if you create it, you delete it.
+        /// </summary>
+        /// <param name="allocator">Should be IntPtr.Zero</param>
+        /// <param name="str">The string to get a CFStringRef for</param>
+        /// <param name="encoding">The encoding of the str variable. This should be UTF 8 for OS X</param>
```

Comment:
(The same could apply to other places where your comment states that only one particular value should be passed.)